src/plots/plots.js

```diff
@@ -1371,11 +1371,10 @@ plots.plotAutoSize = function plotAutoSize(gd, layout, fullLayout) {
             reservedHeight = reservedMargins.bottom + reservedMargins.top,
             factor = 1 - 2 * frameMargins;
 
-        var gdBB = fullLayout._container && fullLayout._container.node ?
-            fullLayout._container.node().getBoundingClientRect() : {
-                width: fullLayout.width,
-                height: fullLayout.height
-            };
+        var gdBB = {
+            width: fullLayout.width,
+            height: fullLayout.height
+        };
 
```

Comment:
@alexcjohnson I changed this part in order to pass the test saying it `should fill the container when autosize: true, fillFrame: false and frameMargins: 0.1`. I'm worried that this was solving an edge case which is not poperly tested as of right now. What do you think?

Reply:
Looks to me as though all of this dates to the toolpanel era (but not the toolpanel itself, AFAICT)... the only place I see a nonzero `frameMargins` is in [super-old streambed code](https://github.com/plotly/streambed/blob/f1734c1ac81bd33482b4d4ea58130242831e7322/shelly/shelly/static/js/tabs.js#L52) and the only place I see `gd._boundingBoxMargins` is embedded in a coverage test I apparently ran in 2015. So I'm tempted to say we should 🔪 this whole section but that's probably not a good idea. We should certainly 🔪 `calculateReservedMargins` and `gd._boundingBoxMargins`

Lets try to fold `frameMargins` into the block below (`// plotly.js - let the developers do what they want`) and see if we can make something meaningful out of that. I think if you get rid of the `getBoundingClientRect`, that was intended to be the primary path so we're losing the purpose here - which seems to be to size the plot to some fixed percentage smaller than its container. Below, the `getBoundingClientRect` was replaced by `window.getComputedStyle` in order to support more cases, so if we're going to keep `frameMargins` it should benefit from that as well.

Reply:
https://github.com/plotly/plotly.js/pull/3056/commits/b3a4f23155a953d780422c9f5e964b4be93b23e3 looks great. Thanks @antoinerg !